sdk/web-pubsub/web-pubsub-express/src/cloudEventsDispatcher.ts

```diff
@@ -225,14 +260,51 @@ export class CloudEventsDispatcher {
     return value[0];
   }
 
-  private GetContext(ce: CloudEvent, origin: string): ConnectionContext {
+  /**
+   * Convert the states object to base64 JSON string
+   * @param states the connection states
+   * @returns The base64 JSON string
+   */
+  private stringifyStates(states: Record<string, string>): string {
+    return Buffer.from(JSON.stringify(states)).toString("base64");
+  }
+
+  /**
+   * Connection state is base64(jsonString)
+   * @param ce
+   * @param context
+   */
+  private parseStates(state: string): Record<string, any> {
+    var states: Record<string, any> = {};
+    if (state === undefined) {
+      return states;
+    }
+
+    try {
+      var buf = Buffer.from(state, "base64").toString();
+      var parsed = JSON.parse(buf);
+      for (const key in parsed) {
+        if (Object.prototype.hasOwnProperty.call(parsed, key)) {
+          states[key] = parsed[key];
+        }
+      }
```

Comment:
why are you cloning the parsed object into `states`? You can just return `parsed` here, right?

Reply:
a valid JSON string `parsed` is not necessarily a key-value pair object, it can also be array or string, I use such format to guarantee that only key-value style JSON object is set to the state. Any suggestions?

Reply:
If you're wanting to validate the type of the parsed JSON, why not something like `if (typeof(parsed) === 'object' && !Array.isArray(parsed)`?  Your above code will end up coercing arrays into objects with properties named after the array indexes, which may not be what you intend?

Reply:
thanks for the suggestion!